### PR TITLE
Create new user groups and invites form

### PR DESCRIPTION
**Pull request by itsliterallymonique**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: types/schema.d.ts [hunk 1]
```diff
@@ -3,9 +3,12 @@ import type { UUID } from "crypto";
 // ENUM for Emails.user_email
 export type EmailType = "PLACEHOLDER";
 
-// ENUM for Profiles.user_group
+// ENUM for user_type / user_group
 export type UserType = "Admin" | "Facilitator" | "Participant";
 
+// ENUM for status
+export type StatusType = "Pending" | "Accepted";
```
Comment (me-liu) [inline types/schema.d.ts]: can you also add another here that says 'Cancelled'

File: app/user-groups/add-user-groups/page.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const [userGroupInput, setUserGroupInput] = useState("");
+  const router = useRouter();
+  const userId = "";
+  const handleInputChange = (event: {
+    target: { value: SetStateAction<string> };
+  }) => {
+    setUserGroupInput(event.target.value);
+  };
+  const onSubmitButtonClick = async (
+    event: React.MouseEvent<HTMLButtonElement>,
```
Comment (me-liu) [inline app/user-groups/add-user-groups/page.tsx]: you can remove this line as it is not being used

File: app/user-groups/add-facilitators/page.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    setFacilitatorEmail(event.target.value);
+  };
+
+  const onAddFacilitatorButtonClick = (
+    event: React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    setAddFacilitator(true);
+  };
+
+  const onSubmitButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (me-liu) [inline app/user-groups/add-facilitators/page.tsx]: the event is unnecessary here too

File: api/supabase/queries/invites.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  invited_by_user_id: string,
+): Promise<void> {
+  const id = crypto.randomUUID();
+  const { data, error } = await supabase.from("invite").upsert(
+    {
+      invite_id: id,
+      user_group_id: user_group_id,
+      //invited_by_user_id: invited_by_user_id,
+      email: email,
+      user_type: "Facilitator",
```
Comment (me-liu) [inline api/supabase/queries/invites.ts]: also can you add a prop in this function called `isFacilitator` and then you use it to specify whether or not to make it a facilitator invite

**Review by me-liu — COMMENTED** [review]
very solid work hooray--minor nits!

File: app/user-groups/add-facilitators/page.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  FacilitatorEmailDiv,
+  FacilitatorEmailInput,
+  SubmitButton,
+} from "./styles";
+
+export default function AddFacilitators() {
+  const searchParams = useSearchParams();
+  const userId = searchParams.get("userId");
+  const userGroupId = searchParams.get("userGroupId");
+  const [addFacilitator, setAddFacilitator] = useState(false);
```
Comment (me-liu) [inline app/user-groups/add-facilitators/page.tsx]: also btw this makes it so that you can only add 1 facilitator at once through by adding 1 input box. The 'add facilitator' button should be adding one input box at a time. the useStates should look something like this: 

```
  const [facilitatorEmails, setFacilitatorEmails] = useState<string[]>([""]);

  const handleInputChange = (index: number, value: string) => {
    const updated = [...facilitatorEmails];
    updated[index] = value;
    setFacilitatorEmails(updated);
  };

  const handleAddFacilitatorClick = () => {
    setFacilitatorEmails([...facilitatorEmails, ""]);
  };
```
you also no longer need this
```
  const [addFacilitator, setAddFacilitator] = useState(false);
```
and then you'd map through these to display them. You'd also iterate through them when calling submitNewFacilitator on each.

Comment (me-liu) [reply app/user-groups/add-facilitators/page.tsx]: Screenshot 2025-10-23 at 1 08 06 AM
so it might look something like this
